tests/test_barostat.py

```diff
@@ -232,8 +232,16 @@ def test_barostat_partial_group_idxs():
     ctxt.multiple_steps(barostat_interval * 100)
 
 
-@pytest.mark.memcheck
-def test_barostat_is_deterministic():
+@pytest.mark.parametrize(
+    "box_width, iterations",
+    [
+        pytest.param(3.0, 20, marks=pytest.mark.memcheck),
+        # fyork: This test only fails 50-50 times. It tests a race condition in which the coordinates could be corrupted.
```

Comment:
Could we make the system bigger or increase the number of iterations to increase the failure rate (assuming this would make a significant difference)? (this only takes ~seconds on my machine)

Reply:
Also, is it possible to trigger the error without running MD in between the barostat moves?

Reply:
> the comment refers to the settings below the comment

oops, sorry, I had misread this and run the wrong set of parameters. Makes sense that we probably don't want to go bigger for a unit test.

Reply:
> Possibly? Though it only triggers on accepted moves, so would have to construct a system that reliably accepts moves.

One idea (not necessary for this PR): scale the coordinates by (e.g. by 0.99) between each barostat move?

Reply:
> > Possibly? Though it only triggers on accepted moves, so would have to construct a system that reliably accepts moves.
> 
> One idea (not necessary for this PR): scale the coordinates by (e.g. by 0.99, tune to get desired acceptance rate) between each barostat move?

Disregard since it sounds like the symptom is a blowup in MD; would need to figure another way to detect the invalid behavior, which is probably out of scope for this PR.